### PR TITLE
proper separation of engine_ and backend

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: engine/keep.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package engine
```
Comment (yperbasis) [inline engine/keep.go:1]: What's the purpose of this file?

File: engine/engine.proto [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  rpc EngineGetPayload(EngineGetPayloadRequest) returns(EngineGetPayloadResponse);
+
+  rpc EngineGetPayloadBodiesByHashV1(EngineGetPayloadBodiesByHashV1Request) returns(EngineGetPayloadBodiesV1Response);
+
+  rpc EngineGetPayloadBodiesByRangeV1(EngineGetPayloadBodiesByRangeV1Request) returns(EngineGetPayloadBodiesV1Response);
+
+  // Fetch the blobs bundle using its ID.
+  rpc EngineGetBlobsBundleV1(EngineGetBlobsBundleRequest) returns(types.BlobsBundleV1);
+  // Should be moved to execution.proto
+  rpc PendingBlock(google.protobuf.Empty) returns (PendingBlockReply);
```
Comment (yperbasis) [inline engine/engine.proto:30]: Why is `PendingBlock` duplicated in engine & execution?

File: engine/keep.go (hunk 1 above)
Comment (Giulio2002) [reply engine/keep.go:1]: same as https://github.com/ledgerwatch/interfaces/blob/master/remote/keep.go

File: engine/engine.proto (hunk 2 above)
Comment (Giulio2002) [reply engine/engine.proto:30]: So ideally PendingBlock should be part of Execution.proto at a conceptual level. However, as of now PendingBlock rely on engine logic. Therefore, I repeated it, it is also a nice way to remind myself to fix it later on the road.

Comment (Giulio2002) [reply engine/engine.proto:30]: Remember: this is just separation of logic, I think the whole Engine GRPC interface is temporary as well as I want to move to json

**Review by yperbasis — APPROVED** [review]

**Pull request merged**